src/components/Firestore/FirestoreEmulatedApiProvider.tsx

```diff
@@ -0,0 +1,141 @@
+import firebase from 'firebase';
+import React, { Suspense, useEffect } from 'react';
+import { FirebaseAppProvider, useFirestore } from 'reactfire';
+import useSwr from 'swr';
+
+import { useEmulatedFirebaseApp } from '../../firebase';
+import { useFirestoreConfig, useProjectId } from '../../store/config/selectors';
+
+interface WindowWithFirestore extends Window {
+  firestore?: firebase.firestore.Firestore;
+}
+
+/**
+ * Provide a local-FirebaseApp with a FirestoreSDK connected to
+ * the Emulator Hub.
+ *
+ * TODO(tlavelle): need loading indicator for FirestoreSDK
```

Comment:
TBH we should just do it. There's a loading layout helper component already

Reply:
Done, <Spinner /> is super nice!